src/components/Dropzone/Dropzone.tsx

```diff
@@ -133,3 +118,68 @@ export const FileList = () => {
     </Flex>
   );
 };
+
+async function processImages(
+  files: File[],
+  onSuccess: (image: ImageFile) => void,
+  onError: (reason: string) => void,
+  onAbort: (reason: string) => void,
+) {
+  for (const file of files) {
+    try {
+      const scaledImage = await scaleImage(file, 800);
+      const fileForChat = {
+        name: file.name,
+        content: scaledImage,
+        type: file.type,
+      };
+
+      onSuccess(fileForChat);
+    } catch (error) {
+      if (error === "abort") {
+        onAbort(`file ${file.name} reading was aborted`);
+      } else {
+        onError(`file ${file.name} processing has failed`);
```

Comment:
Works great with PNG images, but if I do paste TIF image file, it should say an error of mimetype, isn't it?
![image](https://github.com/user-attachments/assets/444aa7af-2a53-461d-940f-427d3ec194b8)

Reply:
hmm looks like `canvas` and `img` tag don't support TIFs,
there is a library to convert them https://github.com/seikichi/tiff.js/blob/master/tiff.js

Reply:
^that library seems to have a lot of issues,
maybe this one https://jimp-dev.github.io/jimp/guides/custom-jimp/